models/user.js

```diff
@@ -1,11 +1,38 @@
 var mongoose = require('./setup');
 
-var Schema = mongoose.Schema,
-    ObjectId = Schema.ObjectId;
-
-var userSchema = new Schema({
+var userSchema = new mongoose.Schema({
   email: { type: String, required: '{PATH} is required!' },
+  firstName: { type: String, required: '{PATH} is required!' },
+  lastName: { type: String, required: '{PATH} is required!' }
 });
 
+userSchema.statics.findOrCreateBefore = function (user, cb) {
+  this.findOne({ email: user.emails[0] }, (err, user) => {
+    if (err) {
+      console.log(err);
+    }
+    else {
+      if (!user) {
```

Comment:
Invert this if statement. 
if (user) {
   cb(user)
   return
}

// rest of logic